File: app/src/main/java/soup/movie/data/soup/model/TrailerResponse.java
package soup.movie.data.soup.model;

import com.google.gson.annotations.JsonAdapter;

import java.util.List;

import soup.movie.data.soup.converter.TrailerResponseDeserializer;

@JsonAdapter(TrailerResponseDeserializer.class)
public class TrailerResponse {

    private List<Trailer> trailerList;

    public TrailerResponse() {
    }

    public List<Trailer> getTrailerList() {
        return trailerList;
    }

    public void setTrailerList(List<Trailer> trailerList) {
        this.trailerList = trailerList;
    }

    @Override
    public String toString() {
        return "TrailerResponse{" +
                "trailerList=" + trailerList +
                '}';
    }
}

File: data/repository/src/main/java/soup/movie/data/model/response/TrailerResponse.kt
package soup.movie.data.model.response

import androidx.core.text.parseAsHtml
import com.google.gson.annotations.SerializedName

data class TrailerResponse(
    val youtubeId: String,
    @SerializedName("title")
    private val _title: String,
    val author: String,
    val thumbnailUrl: String
) {
    val title: String
        get() = _title.parseAsHtml().toString()
}
